src/SubdomainChecker.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class SubdomainChecker
+{
+    private static $initiated = false;
+    private static $isSubdomain = false;
+    private static $originalDomain = '';
+    private static $subDomain = '';
+
+    public static function init()
+    {
+        if (!self::$initiated) {
+            self::init_hooks();
+        }
+
```

Comment:
Is this something all plugins can use?  Should we move it to cloudflare/cloudflare-plugin-backend?